gammapy/estimators/map/ts.py

```diff
@@ -443,10 +484,34 @@ def estimate_flux_map(self, dataset):
         j, i = zip(*positions)
 
         geom = maps["counts"].geom.squash(axis_name="energy")
+        energy_axis = maps["counts"].geom.axes["energy"]
+        dnde_ref = self.model.spectral_model(energy_axis.center)
 
         for name in self.selection_all:
-            m = Map.from_geom(geom=geom, data=np.nan, unit="")
-            m.data[0, j, i] = [_[name] for _ in results]
+            if name in ["dnde_scan_values", "stat_scan"]:
+                norm_bin_axis = MapAxis(
+                    range(len(results[0]["dnde_scan_values"])),
+                    interp="lin",
+                    node_type="center",
+                    name="dnde_bin",
+                )
+
+                axes = geom.axes + [norm_bin_axis]
+                geom_scan = geom.to_image().to_cube(axes)
+
+                if name == "dnde_scan_values":
+                    unit = dnde_ref.unit
+                    factor = dnde_ref.value
+                else:
+                    unit = ""
+                    factor = 1
+
+                m = Map.from_geom(geom_scan, data=np.nan, unit=unit)
+                m.data[:, 0, j, i] = np.array([_[name] for _ in results]).T * factor
```

Comment:
Is the transposition enough here? have you tried with non-square maps?

Reply:
the map from the results object here is always 2d here and it has the dimension (n_valid_pixels, n_scan_values=109)